File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        /**
        people을 내림차순 [80, 70, 50, 50]
        오름차순 [50, 50, 70, 80]
        [10, 20, 20, 40, 60, 80]
        */
        Arrays.sort(people);
        int min = 0;
        for (int i = people.length - 1; i >= min; i--) {
            if (people[i] + people[min] <= limit) {
                answer++;
                min++;
            }
            else {
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    answer = 0
    right = len(people) - 1
    left, count = 0, 0
    
    while left <= right:
        if people[left] + people[right] <= limit:
            left += 1
            right -= 1
        else:
            right -= 1
        
        count += 1
    
    return count